src/layers/layer.cpp

```diff
@@ -105,7 +105,7 @@ description Layer::get_description() const {
 
   // Construct description object
   std::stringstream ss;
-  ss << get_name() << " (" << get_type() << ")";
+  ss << get_name() << " (" << get_type() << "<" << get_datatype_name() << ">" << ")";
```

Comment:
This produces output like:
```
    layer15 (softmax<float>)
      Input dimensions: 10 (from fully connected layer "layer14")
      Output dimensions: 10 (to split layer "layer15_split")
      Data layout: data-parallel
      Device: GPU
```
I'd prefer something like:
```
    layer15 (softmax)
      Input dimensions: 10 (from fully connected layer "layer14")
      Output dimensions: 10 (to split layer "layer15_split")
      Data layout: data-parallel
      Device: GPU
      Data type: float
```
I think this has several advantages:
- The header is less cluttered.
- It allows us to move `get_datatype_name` out of the base `layer` class to `data_type_layer`, which is logically cleaner. `data_type_layer` can override `get_description` to add the datatype info.

Reply:
Done